controllers/postgresqldatabase_controller.go

```diff
@@ -103,7 +105,11 @@ func (r *PostgreSQLDatabaseReconciler) reconcile(reqLogger logr.Logger, request
 	}
 	host, err := kube.ResourceValue(r.Client, database.Spec.Host, request.Namespace)
 	if err != nil {
-		return status, fmt.Errorf("resolve host reference: %w", err)
+		// if the `host` value is missing, we want to keep going because it
+		// should mean that the `hostCredentials` is provided.
+		if !errors.Is(err, kube.ErrNoValue) {
+			return status, fmt.Errorf("resolve host reference: %w", err)
+		}
 	}
 	status.host = host
```

Comment:
I'd like us to resolve the `hostCredentials` field here instead of inside the `EnsurePostgreSQLDatabase` method. The reasoning being that the `EnsurePostgreSQLDatabase` should only ensure the database not figure out if the client has provided all the data. It also aligns with the rest of this method's responsibility being "resolve all references to ensure we are in a good state" before it hands it off.

Maybe we can just move the call to `r.credentials` up here instead. We could then move the `ResourceValue` call above on `host` into that method to have "everything credentials" in the new method. WDYT?

Reply:
Yeah, this makes sense. I didn't pick up on the intended contract between `EnsurePostgreSQLDatabase` and its caller. 👍

I'll take a stab at it and you can let me know if that matches what your intention.

Reply:
This should be addressed via 7a3ea571866fcdd6a6f2afc69d1d0475b935c17e.

Reply:
It wasn't clear at all from neither the naming or the non-existing docs 🙈